updater/lib/dependabot/dependency_snapshot.rb

```diff
@@ -53,6 +53,18 @@ def job_dependencies
       end
     end
 
+    # A dependency snapshot will always have the same set of dependencies since it only depends
+    # on the Job and dependency groups, which are static for a given commit.
+    def groups
+      # The DependencyGroupEngine registers dependencies when the Job is created
+      # and it will memoize the dependency groups
+      Dependabot::DependencyGroupEngine.dependency_groups(allowed_dependencies)
```

Comment:
```suggestion
      @groups ||= Dependabot::DependencyGroupEngine.dependency_groups(allowed_dependencies)
```

Should we memoize this to avoid naively recalculating it?

Reply:
Disregard this, we should only call this once and its effectively hitting a singleton class so memoization isn't really useful.